File: DATA/Boxscores/LIBRARY/TeamsClass.R
library(RSQLite)
library(DBI)
library(S7)
library(dplyr)


Teams <- new_class(
  "Teams",
  properties = list(
    path = new_property(class_character)
  ),
  # Custom constructor to set the path property
  constructor = function(path_to_database = character()) {
    new_object(S7_object(), path = path_to_database)
  }
)

# Create a new generic function for retrieving the data frame
teams_dataframe <- new_generic("teams_dataframe", "x")

method(teams_dataframe, Teams) <- function(x) {

  con <- dbConnect(RSQLite::SQLite(), x@path)
  
  query <- tbl(con, "Teams")
    
  results_df <- query |>
    collect()
    
  dbDisconnect(con)
    
    return(results_df)
}

get_Teams_instance <- function(db_path) {
  Teams(path_to_database = db_path)
}

File: DATA/Boxscores/LIBRARY/TeamsClass.py
import pandas as pd
import sqlite3

class Teams:

    path: str
    
    def __init__(self, path_to_database: str = ""):
       
        self.path = path_to_database
    
    def teams_dataframe(self) -> pd.DataFrame:
      
        # Connect to database
        con = sqlite3.connect(self.path)
        
        try:
            # Query the Teams table
            query = "SELECT * FROM Teams"
            results_df = pd.read_sql_query(query, con)
            
            return results_df
        
        finally:
            # Always close the connection
            con.close()


def get_Teams_instance(db_path: str) -> Teams:
 
    return Teams(path_to_database=db_path)

